python/ray/tests/test_joblib.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import joblib
+from sklearn.datasets import load_digits, load_iris
+from sklearn.model_selection import RandomizedSearchCV
+from time import time
+from sklearn.datasets import fetch_openml
+from sklearn.ensemble import ExtraTreesClassifier
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.kernel_approximation import Nystroem
+from sklearn.kernel_approximation import RBFSampler
+from sklearn.pipeline import make_pipeline
+from sklearn.svm import LinearSVC, SVC
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.utils import check_array
+from sklearn.linear_model import LogisticRegression
+from sklearn.neural_network import MLPClassifier
+from sklearn.model_selection import cross_val_score
+
+from ray.experimental.joblib import register_ray
+import ray
+
+
+def test_register_ray():
+    register_ray()
+    assert "ray" in joblib.parallel.BACKENDS
+    assert not ray.is_initialized()
+
+
+def test_ray_backend(shutdown_only):
+    register_ray()
+    from ray.experimental.joblib._raybackend import RayBackend
+    with joblib.parallel_backend("ray"):
+        assert type(joblib.parallel.get_active_backend()[0]) == RayBackend
+
+
+def test_svm_single_node(shutdown_only):
+    digits = load_digits()
+    param_space = {
+        "C": np.logspace(-6, 6, 10),
+        "gamma": np.logspace(-8, 8, 10),
+        "tol": np.logspace(-4, -1, 3),
+        "class_weight": [None, "balanced"],
+    }
+
+    model = SVC(kernel="rbf")
+    search = RandomizedSearchCV(
+        model, param_space, cv=3, n_iter=50, verbose=10)
+    register_ray()
+    with joblib.parallel_backend("ray"):
+        search.fit(digits.data, digits.target)
+    assert ray.is_initialized()
+
+
+def test_svm_multiple_nodes(ray_start_cluster_2_nodes):
+    digits = load_digits()
+    param_space = {
+        "C": np.logspace(-6, 6, 30),
+        "gamma": np.logspace(-8, 8, 30),
+        "tol": np.logspace(-4, -1, 30),
+        "class_weight": [None, "balanced"],
+    }
+
+    model = SVC(kernel="rbf")
+    search = RandomizedSearchCV(
+        model, param_space, cv=5, n_iter=100, verbose=10)
+    register_ray()
+    with joblib.parallel_backend("ray"):
+        search.fit(digits.data, digits.target)
+    assert ray.is_initialized()
+
+
+def test_sklearn_benchmarks(ray_start_cluster_2_nodes):
```

Comment:
Not sure where the other comment went, but I don't think printing a warning is sufficient for a test case. This should explicitly pass or fail on some assertion clause. Otherwise, we can just make it a benchmark but it shouldn't be run as part of the tests.